win32ss/user/ntuser/message.c

```diff
@@ -2537,7 +2537,10 @@ NtUserMessageCall( HWND hWnd,
     PWND Window = NULL;
     USER_REFERENCE_ENTRY Ref;
 
-    UserEnterExclusive();
+    if (dwType == FNID_DEFWINDOWPROC && Msg == WM_POPUPSYSTEMMENU)
```

Comment:
What would happen if you were using UserEnterExclusive also for WM_POPUPSYSTEMMENU?